src/App.tsx

```diff
@@ -1,61 +1,121 @@
 import './App.scss';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  // if there is no user with a given userId
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+let timeId = Math.max(...todos.map(todo => todo.id));
+
+const newTodo = (title: string, userId: number): Todo => ({
+  id: timeId,
+  userId,
+  title,
+  completed: false,
+  user: getUser(userId),
+});
+
+const addTodo = (title: string, id: number) => {
+  timeId += 1;
+  const todo = newTodo(title, id);
+
+  todos.push(todo);
+};
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [user, setUser] = useState('');
```

Comment:
since you set only id, better name it `userId, setUserId`

Reply:
changed